staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -33,21 +33,42 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/conversion"
 	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/apiserver/pkg/storage"
-	utiltrace "k8s.io/apiserver/pkg/util/trace"
 	"k8s.io/apiserver/pkg/storage/etcd"
+	utiltrace "k8s.io/apiserver/pkg/util/trace"
 )
 
+// ValueTransformer allows a string value to be transformed before being read from or written to the underlying store. The methods
+// must be able to undo the transformation caused by the other.
+type ValueTransformer interface {
```

Comment:
I was hoping we'd drop etcd2 first.  For now, I'd like to avoid unnecessary copies on every store (etcd3 is forcing it), but I would expect that someone would implement both interfaces as efficiently as possible.